app/controllers/admin/collections_controller.rb

```diff
@@ -12,11 +12,22 @@ def index
     # Searching, filtering, sorting and pagination.
     scope = Collection
     if index_params[:title_or_id].present?
+      sanitized_search_phrase = Collection.sanitize_sql_like(params[:title_or_id])
+      # This expression will match any element of the external_id jsonb array,
+      # as long as its value is sanitized_search_phrase, regardless of the value of 'category'.
+      jsonb_match_value = "'[{\"value\": \"#{sanitized_search_phrase}\"}]'::jsonb"
+
+      # This condition attempts to match external_id against jsonb_match_value:
+      matches_external_id = "json_attributes -> 'external_id' @> #{jsonb_match_value}"
+
+
       scope = scope.where(id: index_params[:title_or_id]
       ).or(
-        Collection.where(friendlier_id: index_params[:title_or_id])
+        scope.where(friendlier_id: index_params[:title_or_id])
+      ).or(
+        scope.where("title ilike ?", "%" + sanitized_search_phrase + "%")
       ).or(
-        Collection.where("title ilike ?", "%" + Collection.sanitize_sql_like(index_params[:title_or_id]) + "%")
+        scope.where(matches_external_id)
       )
```

Comment:
I _think_ this actually should be `Collection.where` instead of `scope.where`?  They may be equivalent right now (or not), but as we add things onto scope, they could creep into the `or` clauses, when each or clauses meant to be independent and start from `Collection` I think?

I could be missing something!

Reply:
You're right - thanks for catching this!